internal/integration/csot_prose_test.go

```diff
@@ -177,6 +177,63 @@ func TestCSOTProse(t *testing.T) {
 				"expected ping to fail within 150ms")
 		})
 	})
+
+	mt.RunOpts("11. multi-batch bulkWrites", mtest.NewOptions().MinServerVersion("8.0").
+		AtlasDataLake(false).Topologies(mtest.Single), func(mt *mtest.T) {
+		coll := mt.CreateCollection(mtest.Collection{DB: "db", Name: "coll"}, false)
+		err := coll.Drop(context.Background())
+		require.NoError(mt, err, "Drop error: %v", err)
+
+		mt.SetFailPoint(failpoint.FailPoint{
+			ConfigureFailPoint: "failCommand",
+			Mode: failpoint.Mode{
+				Times: 2,
+			},
+			Data: failpoint.Data{
+				FailCommands:    []string{"bulkWrite"},
+				BlockConnection: true,
+				BlockTimeMS:     1010,
+			},
+		})
+
+		var hello struct {
+			MaxBsonObjectSize   int
+			MaxMessageSizeBytes int
+		}
+		err = mt.DB.RunCommand(context.Background(), bson.D{{"hello", 1}}).Decode(&hello)
+		require.NoError(mt, err, "Hello error: %v", err)
```

Comment:
Optional: This block is repeated a lot in the tests and is not part of the code-under-test, so is a good candidate for a helper.

E.g.
```go
func getServerMaxSizes(mt *mtest.T) (maxBSONObjectSize, maxMessageSize int) {
	mt.Helper()
	// ...
```

Reply:
I was considering making it a helper. However, each of them requires a slightly different set of max values, all in the integer type, leading to the return of the helper in a lengthy and baffling list of integers. Moreover, the core logic is only a single function call, so making it a helper does not necessarily make the code neat.

Reply:
Sounds good.